pkg/csplugin/broker_test.go

```diff
@@ -113,8 +133,175 @@ func Test_listFilesAtPath(t *testing.T) {
+			errContains: "both plugin user and group must be set",
+			procCfg: csconfig.PluginCfg{
+				User: "123445555551122toto",
+			},
+			action: setPluginPermTo744,
+		},
+		{
+			name:        "only specify group",
+			wantErr:     true,
+			errContains: "both plugin user and group must be set",
+			procCfg: csconfig.PluginCfg{
+				Group: "123445555551122toto",
+			},
+			action: setPluginPermTo744,
+		},
+		{
+			name:        "Fails to run as root",
+			wantErr:     true,
+			errContains: "operation not permitted",
+			procCfg: csconfig.PluginCfg{
+				User:  "root",
+				Group: "root",
+			},
+			action: setPluginPermTo744,
+		},
+		{
+			name:        "Invalid user and group",
+			wantErr:     true,
+			errContains: "unknown user toto1234",
+			procCfg: csconfig.PluginCfg{
+				User:  "toto1234",
+				Group: "toto1234",
+			},
+			action: setPluginPermTo744,
+		},
+		{
+			name:        "Valid user and invalid group",
+			wantErr:     true,
+			errContains: "unknown group toto1234",
+			procCfg: csconfig.PluginCfg{
+				User:  "nobody",
+				Group: "toto1234",
+			},
+			action: setPluginPermTo744,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			defer tearDown()
+			buildDummyPlugin()
+			if test.action != nil {
+				test.action()
+			}
+			pb := PluginBroker{}
+			profiles := csconfig.NewDefaultConfig().API.Server.Profiles
+			profiles = append(profiles, &csconfig.ProfileCfg{
+				Notifications: []string{"dummy_default"},
+			})
+			err := pb.Init(&test.procCfg, profiles, &csconfig.ConfigurationPaths{
+				PluginDir:       testPath,
+				NotificationDir: "./tests/notifications",
+			})
+			defer pb.Kill()
+			if test.wantErr {
+				assert.ErrorContains(t, err, test.errContains)
+			} else {
+				assert.NoError(t, err)
+			}
+
+		})
+	}
+}
+
+func TestBrokerRun(t *testing.T) {
+	buildDummyPlugin()
+	setPluginPermTo744()
+	defer tearDown()
+	procCfg := csconfig.PluginCfg{}
+	pb := PluginBroker{}
+	profiles := csconfig.NewDefaultConfig().API.Server.Profiles
+	profiles = append(profiles, &csconfig.ProfileCfg{
+		Notifications: []string{"dummy_default"},
+	})
+	err := pb.Init(&procCfg, profiles, &csconfig.ConfigurationPaths{
+		PluginDir:       testPath,
+		NotificationDir: "./tests/notifications",
+	})
+	assert.NoError(t, err)
+	tomb := tomb.Tomb{}
+	go pb.Run(&tomb)
+	defer pb.Kill()
+
+	assert.NoFileExists(t, "./out")
+	defer os.Remove("./out")
+
+	pb.PluginChannel <- ProfileAlert{ProfileID: uint(0), Alert: &models.Alert{}}
+	pb.PluginChannel <- ProfileAlert{ProfileID: uint(0), Alert: &models.Alert{}}
+	time.Sleep(time.Second * 4)
+
+	assert.FileExists(t, "./out")
+	assert.Equal(t, types.GetLineCountForFile("./out"), 2)
+}
+
+func buildDummyPlugin() {
+	dir, err := ioutil.TempDir("./tests", "cs_plugin_test")
+	if err != nil {
+		log.Fatal(err)
+	}
+	cmd := exec.Command("go", "build", "-o", path.Join(dir, "notification-dummy"), "../../plugins/notifications/dummy/")
+	if err := cmd.Run(); err != nil {
```

Comment:
Why testing the built executable and not test its code ?

Reply:
I guess the point here is to test the broker, not a specific plugin